Move generated protos into oak_abi

daviddrysdale opened the pull request:
Protocol buffer messages that are exchanged in serialized form between
Wasm Nodes and Oak-provided pseudo-Nodes are effectively part of the
Oak ABI, so move the generated code for them into the oak_abi crate.

This allows us to remove the oak_runtime->oak dependency that was
introduced by #772.

Along the way, put the gRPC encapsulation protobuf messages into an
`encap` submodule/namespace.

Fixes #764.

# Checklist

- [x] Pull request affects core Oak functionality (e.g. runtime, SDK, ABI)
  - [x] I have written tests that cover the code changes.
  - [ ] I have checked that these tests are run by [Cloudbuild](cloudbuild.yaml)
  - [x] I have updated [documentation](docs/) accordingly.
  - [ ] I have raised an [issue](https://github.com/project-oak/oak/issues) to
        cover any TODOs and/or unfinished work.
- [ ] Pull request includes prototype/experimental work that is under
      construction.

daviddrysdale commented:
I'm in two minds as to whether to keep the last two commits.  On the one hand, it feels more complete to include the storage channel messages in `oak_abi`, as they form part of the ABI.  On the other hand, the only code to use them (the `oak::storage` SDK code) needs its own copy anyway (because it needs the full service definition, which can't be included in `oak_abi` because it assumes the use of the `oak` SDK crate).

tiziano88 commented on sdk/rust/oak/src/lib.rs:
```diff
@@ -31,26 +31,13 @@ pub mod rand;
-        }
-    }
     pub mod oak {
+        use oak_abi::proto::oak::label;
```
Maybe add a comment to clarify why this re-export is needed? Presumably we would like to make it so that developers only need to depend on `oak` and not `oak_abi` directly? I guess that makes sense, though I'm always a bit hesitant to introduce re-exports, as having multiple names for the same things can be confusing.

daviddrysdale replied:
Added a comment &ndash; the storage message definitions rely on the `Label` message, which is now (only) generated under the `oak_abi` crate.

tiziano88 commented on oak/proto/storage_channel.proto:
```diff
@@ -16,73 +16,9 @@
-
-message StorageChannelRollbackResponse {
-}
+import "oak/proto/storage_messages.proto";
```
Not sure I follow what's going on here. Why is the service definition not part of the ABI? It seems it should be.

daviddrysdale replied:
The generated code for the service definition assumes the use of the Oak SDK code, and so cannot live in the `oak_abi` crate.  That leads to the separation of the message definitions (which only depend on `prost` and so can live in `oak_abi`) from the service definition generated code (still in `oak`, but now with a parallel copy of the message definitions), which leads to my [earlier question](https://github.com/project-oak/oak/pull/834#issuecomment-611552052) &ndash; is it worth bothering with the storage protos?

tiziano88 commented on oak/proto/storage_channel.proto:
```diff
@@ -16,73 +16,9 @@
 
```
perhaps this file should be renamed to `storage_node` or `storage_service` at some point? (Or just deleted -- hint hint)

daviddrysdale replied:
Seems reasonable, but for another day.

daviddrysdale replied:
Actually, done: it's now `StorageService` in `oak/proto/storage_service.proto`

daviddrysdale commented:
Dropped the last two commits (which split the storage proto and copied the message definitions to `oak_abi`) and added a rename commit for storage_channel.proto.
